lib/sqlite-manager.js

```diff
@@ -9,1031 +9,1017 @@
 
-    if (tdxSchema.length) {
-      // Dataset schema definition
-      tdxSchema[0].schema = tdxSchema[0].schema || {};
+    // Dataset data schema
+    tdxSchema[0].schema.dataSchema = tdxSchema[0].schema.dataSchema || {};
 
-      // Dataset data schema
-      tdxSchema[0].schema.dataSchema = tdxSchema[0].schema.dataSchema || {};
+    thisGeneralSchema = await sqliteConverter.convertSchema(
+      tdxSchema[0].schema.dataSchema);
 
-      thisGeneralSchema = await sqliteConverter.convertSchema(
-        tdxSchema[0].schema.dataSchema);
+    // Assign the general schema if the info table exists and the dataset exists
+    // eslint-disable-next-line require-atomic-updates
+    db.generalSchema = thisGeneralSchema;
+  }
+  return thisGeneralSchema;
+};
 
-      // Assign the general schema if the info table exists and the dataset exists
-      // Will be empty otherwise
-      db.generalSchema = thisGeneralSchema;
+/**
+ * Add data to a dataset resource.
+ * @function
+ * @async
+ * @alias module:sqlite-manager.addData
+ * @param {object} db - The sqlite3 db object from module node-sqlite3.
+ * @param {DataRow|DataRow[]} data - The data to add.
+ *     Must conform to the schema defined by the resource metadata.
+ *     Supports creating an individual document or many documents.
+ * @return  {Promise<object<string, int>>}
+ *     - The promise with the total count of rows added.
+ * @example <caption>create an individual document</caption>
+ * // returns {"count": 1} if successful
+ * manager.addData(db, {lsoa: "E0000001", count: 398});
+ * @example <caption>create multiple documents</caption>
+ * manager.addData(db, [
+ *  {lsoa: "E0000001", count: 398},
+ *  {lsoa: "E0000002", count: 1775},
+ *  {lsoa: "E0000005", count: 4533},
+ * ]);
+ * @example <caption>add a 2D ndarray</caption>
+ * buffer = Buffer.alloc(23*34);
+ * manager.addData(db, {id: 1, array: manager.getNdarrayMeta(buffer, "float64", [23, 34])});
+ */
+module.exports.addData = async function(db, data) {
+  const schema = await module.exports.getGeneralSchema(db);
+  const dataToConvert = [].concat(data);
+
+  // remove extra columns in data that aren't in the schema
+  const onlySchemaColumns = dataToConvert.map((row) => {
+    const onlySchemaColumnRow = pick(row, Object.keys(schema));
+    if (Object.keys(row).length !== Object.keys(onlySchemaColumnRow).length) {
+      warn(`${"Ignoring extra fields in row that were not in schema. " +
+        `Row was: ${JSON.stringify(row)} and extra keys were: `}${
+        Object.keys(row).filter((col) => schema[col] === undefined)}`,
+      "ExtraFields", "AddDataExtraFields",
+      );
     }
-    return thisGeneralSchema;
+    return onlySchemaColumnRow;
+  });
+
+  // Get the ndarray keys
+  const ndarrayKeys = findCollectionKeys(
+    schema, sqliteConstants.SQLITE_GENERAL_TYPE_NDARRAY);
+
+  // Save the ndarray data to file
+  let arrayProcData = onlySchemaColumns;
+  if (ndarrayKeys. length > 0)
```

Comment:
![image](https://user-images.githubusercontent.com/1726497/67941869-7964e180-fbce-11e9-9879-497937eeddb1.png)

Reply:
Added the eslint rule for this. Again surprised that `eslint:recommended` didn't already contain it.
Good catch!